filebeat/input/v2/compat/compat.go

```diff
@@ -146,7 +153,11 @@ func configID(config *conf.C) (string, error) {
 	}
 
 	var h map[string]interface{}
-	config.Unpack(&h)
+	err := config.Unpack(&h)
+	if err != nil {
+		return "", fmt.Errorf("can not compute id from configuration: unpack failed: %w", err)
```

Comment:
There are other reasons `config.Unpack` could fail here:
```suggestion
		return "", fmt.Errorf("cannot parse configuration: unpack failed: %w", err)
```

Reply:
It does not change the fact that the ID could not be extracted from the configuration. That way whoever reads the error know what was tried and the root cause, thus having more information to understand what is happening.